src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,125 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double SIZE_MULTIPLIER = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elements;
+    private int size = 0;
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            elements = new Object[capacity];
+        } else if (capacity == 0) {
+            elements = new Object[DEFAULT_CAPACITY];
+        } else {
+            throw new IllegalArgumentException("Invalid capacity");
+        }
+    }
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elements.length) {
+            elements = grow();
+        }
```

Comment:
1) you call method grow only covered with this condition.
move condition inside method
2) there is no need to use assignment to class variable here, grow method may be void and change class variable directly
```suggestion
        growIfFull();
```